### PR TITLE
CI for TestPyPI and PyPI

### PR DESCRIPTION
**Pull request by adku1173**
* uses hatch to build wheel and tarball
* adds CI workflows to build and deploy to TestPyPI and PyPI 
* fixes some typos and problems with links and images in README.md
* successful test publishing on TestPyPI -> https://test.pypi.org/project/acoular/ (I had to temporarily use a different version number, because after the upload the respective version is blocked and cannot be used again)

One thing I wasn't sure about was how we should handle it is what the tarball archive needs to include. Probably, the package could be smaller if we exclude some directories such as the tests directory for example. 

One flaw is that the installation with testpypi does not work out of the box, as some package distributions that Acoular depends on are not on TestPyPI. Therefore I had to check locally if the wheels and tarball are correct.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -54,8 +54,8 @@ full = [
 ]
 
 [project.urls]
-homepage = "https://acoular.org"
```
Comment (esarradj) [inline pyproject.toml:57]: https is correct IMHO

File: pyproject.toml [hunk 2]
```diff
@@ -101,3 +101,13 @@ coverage = ["test --cov=acoular"]
 platform.linux.pre-install-commands = ['cat /proc/cpuinfo']
 platform.macos.pre-install-commands = ['sysctl -a machdep.cpu']
 platform.windows.pre-install-commands = ['systeminfo']
+
+
+[tool.hatch.build.targets.sdist]
+exclude = [
```
Comment (esarradj) [inline pyproject.toml:107]: Only what is needed to build the package should be included in the sdist. I think that is just ./acoular, pyproject.toml and some other files in the project root (approx 150 kB). The last sdist on pypi includes everything including test data and is stunning 7 MB large.

Comment (adku1173) [reply pyproject.toml:107]: We handled it differently across various releases (24.03, 23.11, 23.06 -> approx. 7 MB, 22.03, 21.05, ... < 1 MB).

File: pyproject.toml (hunk 1 above)
Comment (adku1173) [reply pyproject.toml]: fixed

File: pyproject.toml (hunk 2 above)
Comment (adku1173) [reply pyproject.toml]: now includes only the source directory

**Review by esarradj — APPROVED** [review]

**Pull request merged**